src/auth/oidc/models/openid.class.ts

```diff
@@ -141,6 +126,39 @@ export class OpenID extends AuthStrategy {
     public useStrategy = (strategyName: string, strategy: Strategy<any, any>): void => {
         passport.use(strategyName, strategy)
     }
+
+    public createNewStrategy = async (options: OpenIDMetadata): Promise<Strategy<any, any>> => {
+        const redirectUri = new URL(AUTH.ROUTE.OAUTH_CALLBACK, options.redirect_uri)
+        this.issuer = await this.discover()
+        if (!this.issuer) {
+            throw new Error('auto discovery failed')
+        }
+        this.client = this.getClientFromIssuer(this.issuer, options)
+        if (!this.client) {
+            throw new Error('client not initialised')
+        }
+        return this.getNewStrategy(redirectUri, options, this.client)
+    }
+
+    public getNewStrategy = (redirectUri: URL, options: OpenIDMetadata, client: Client): Strategy<any, Client> => {
+        return new Strategy(
+            {
+                client,
+                params: {
+                    prompt: OIDC.PROMPT,
+                    // eslint-disable-next-line @typescript-eslint/camelcase
+                    redirect_uri: redirectUri.toString(),
+                    scope: options.scope,
+                },
+                sessionKey: options.sessionKey,
+            },
+            this.verify,
+        )
+    }
+
+    public getClientFromIssuer = (issuer: Issuer<Client>, options: OpenIDMetadata): Client | undefined => {
```

Comment:
You can change this into a one liner if you want, (no return needed) ie. ... : Client | undefined => new issuer.Client(options)

Reply:
Ok, done.